lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_generate_po_file_from_metadata.rb

```diff
@@ -0,0 +1,144 @@
+        @release_version = params[:release_version]
+        @other_sources = params[:other_sources]
+
+        prefix = 'app_store'
+        all_keys = Dir[File.join(@metadata_directory, '*.txt')]
+
+        # Remove from all_keys the special keys as they need to be treated specially
+        standard_files = []
+        all_keys.each do |key|
+          standard_files.append(key) unless SPECIAL_KEYS.include? File.basename(key, '.txt')
+        end
+        # Let the helper handle standard files
+        po = PoExtended.new(:msgctxt)
+        po = Fastlane::Helper::GeneratePoFileMetadataHelper.add_header_to_po(po)
+        po = Fastlane::Helper::GeneratePoFileMetadataHelper.add_standard_files_to_po(prefix, files: standard_files, keys_to_comment_hash: KEYS_TO_COMMENT_HASH, po_obj: po)
+
+        other_sources_files = []
+        @other_sources.each do |other_source|
+          other_sources_files.append(Dir[File.join(other_source, '*.txt')]).flatten!
+        end
+        po = Fastlane::Helper::GeneratePoFileMetadataHelper.add_standard_files_to_po(prefix, files: other_sources_files, keys_to_comment_hash: KEYS_TO_COMMENT_HASH, po_obj: po)
+
+        # Now handle release_notes.txt
+        release_notes_file = File.join(@metadata_directory, 'release_notes.txt')
+        po = Fastlane::Helper::GeneratePoFileMetadataHelper.add_release_notes_to_po(release_notes_file, @release_version, prefix, po, keys_to_comment_hash: KEYS_TO_COMMENT_HASH)
+
+        # Handle release_notes_previous.txt
+        release_notes_previous_file = File.join(@metadata_directory, 'release_notes_previous.txt')
+        version_minus_one = Fastlane::Helper::Ios::VersionHelper.calc_prev_release_version(@release_version)
+        po = Fastlane::Helper::GeneratePoFileMetadataHelper.add_release_notes_to_po(release_notes_previous_file, version_minus_one, prefix, po, keys_to_comment_hash: KEYS_TO_COMMENT_HASH)
+
+        # Finally dump the po into PlayStoreStrings.po
+
+        File.write(File.join(@metadata_directory, 'AppStoreStrings.po'), po.to_s)
+      end
+
+      def self.description
+        'Create the .po based on the .txt files in metadata_directory'
+      end
+
+      def self.details
+        'You can use this action to update the .po file that contains the string to load to GlotPress for localization.'
+      end
+
+      def self.available_options
+        # Define all options your action supports.
+
+        env_name_prefix = 'FL_GENERATE_PO_FILE_FROM_METADATA'
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :metadata_directory,
+            env_name: "#{env_name_prefix}_METADATA_DIRECTORY",
+            description: 'The path containing the .txt files ',
+            is_string: true,
+            verify_block: proc do |value|
+              UI.user_error!("No metadata_directory path for AnGeneratePoFileFromMetadataAction given, pass using `metadata_directory: 'directory'`") unless value && !value.empty?
```

Comment:
> * This type of check could typically be done by fastlane itself just by setting the `optional: false` parameter on your `FastlaneCore::ConfigItem` (just like [in this other action for example](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/lib/fastlane/plugin/wpmreleasetoolkit/actions/common/extract_release_notes_for_version_action.rb#L69)).

Thanks! The less code the happier we are.
PS: `optional` defaults to `false` but it's good to keep it for readability. 
✅ with f98db787b08f5162d047e15626dc629ede9647b2

> * Due likely to some copy/pasting you also have a typo there, as this is the action for iOS but you're mentioning `AnGeneratePoFileFromMetadataAction` (which is even the old class name for Android before you renamed it…)

✅ I just removed them.
> * Besides, the caller who will use this action from their `Fastfile` will not really know about the class name used to implement the action. So if you want to mention the action, it's usually nicer to use the name that will be used in the `Fastfile` to call it, i.e. `ios_generate_po_file_from_metadata`, rather than the class name.
> * Finally, when fastlane errors in a `verify_block`, I think it already mentions the name of the action and of the param (`ConfigItem`) it failed on.


 ✅ I will just omit it then, thanks for bringing this up!